lib/spack/spack/config.py

```diff
@@ -763,6 +768,22 @@ def _add_platform_scope(
     cfg.push_scope(scope_type(plat_name, plat_path))
 
 
+def _add_configuration_paths_from_entry_points(configuration_paths):
+    if importlib_metadata is None:
+        return
+    try:
+        entry_points = importlib_metadata.entry_points(group="spack.config")
+    except TypeError:
+        entry_points = importlib_metadata.entry_points().get("spack.config")
```

Comment:
I'm not super familiar with entry points. Can you explain when we'll get the "try" succeeding, and when we fall back on the exceptional case?

Reply:
@alalazo - the exceptional case is for Python 3.8, the default case is for 3.9 and newer.